ReactiveCocoa/Swift/EventLogger.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  EventLogger.swift
+//  ReactiveCocoa
+//
+//  Created by Rui Peres on 30/04/2016.
+//  Copyright © 2016 GitHub. All rights reserved.
+//
+
+import Foundation
+
+public enum LoggingEvent: String {
+	case Started, Next, Completed, Failed, Terminated, Disposed, Interrupted
+	
+	public static let allEvents: Set<LoggingEvent> = {
+		return [.Started, .Next, .Completed, .Failed, .Terminated, .Disposed, .Interrupted]
+	}()
+}
+
+private func printLog(event: String, fileName: String, functionName: String, lineNumber: Int) -> Void {
+	print("\(event) fileName: \(fileName), functionaName: \(functionName), lineNumber: \(lineNumber)")
+}
+
+public typealias EventLogger = (event: String, fileName: String, functionName: String, lineNumber: Int) -> Void
```

Comment:
> Would it make sense to pass a LoggingEvent and a generic Value? here instead of a String?

the event is fine, the problem is with generic value, because we are using the `printLog` as a default value in the `logEvents` func.